src/components/layout/AddToCartButton/AddToCartButton.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { addItem, reduceItem } from '../../../redux/shoppingCart';
+import { AuthStatus, useAuth } from '@akosasante/react-auth-context';
+import { useNavigate } from 'react-router-dom';
+import Button from '@mui/material/Button';
+
+const AddToCartButton = ({ album }) => {
+  const dispatch = useDispatch();
+  const { status } = useAuth();
+  const navigate = useNavigate();
+  const isUserLoggedIn = status === AuthStatus.LoggedIn;
+  const itemsInCart = useSelector((state) => state.cart.items);
+
+  const handleAddToCart = () => {
+    if (!isUserLoggedIn) {
+      navigate('/signIn');
+      return;
+    }
+
+    if (itemsInCart[album.id]) {
+      // Album is already in the cart, so remove it
+      dispatch(reduceItem({ album: album, quantity: 1 }));
+    } else {
+      // Album is not in the cart, so add it
+      dispatch(addItem({ album: album, quantity: 1 }));
+    }
+  };
+
+  return (
+    <Button
+      variant="contained"
+      color="primary"
+      className="buy-button"
+      aria-label={itemsInCart[album.id] ? 'Remove from cart' : 'Add to Cart'}
+      onClick={handleAddToCart}
+      disabled={!isUserLoggedIn}
```

Comment:
since this is disabled when I'm not logged in; I can't actually click it to get redirected to `signIn`. A couple options I can think of here:

*   don't disable it; and just let it redirect users when they click it. only downside here is it can be a bit surprising for the user when they click that and it immediately sends them away from this page. But otherwise I think it's okay and we can tweak the ux for that later. 
*   disable but show a popover/tooltip when they hover over the button that directs them to sign in. Some notes for this approach, you need to wrap the Button element in a span according to the Material-UI docs for the Tooltip component to work on disabled elements: [https://mui.com/material-ui/react-tooltip/#disabled-elements.](https://mui.com/material-ui/react-tooltip/#disabled-elements.) As well, if you want to use HTML inside the `Tooltip` title prop, you'll need to use a React Fragment for the title instead of a plain string. When the button is not disabled (user is logged in) you can set the Tooltip's title to an empty string and then it won't show up

Reply:
Good point! Thanks for providing a few options how to solve this problem. I chose the first option. The changes were made with the last commit.